### PR TITLE
feat: Integrate shop (buy cat)

### PR DESCRIPTION
**Pull request by hoanghongminh**
Integrate shop - buy cat item

### CODE REVIEW

File: src/components/panels/shop/Shop.tsx [hunk 1]
```diff
@@ -4,33 +4,48 @@ import { useCatDealStore } from "@/stores/shop/catDealStore";
...
+      state.currentCatDeal,
+      state.setCurrentCatDeal,
+      state.setCatDeals,
+    ]);
   const [showCardInfo, setShowCardInfo] = useState(false);
+  const [staff, setStaffs, setCurrentStaff] = useStaffStore((state) => [
+    state.currentStaff,
+    state.setStaffs,
+    state.setCurrentStaff,
+  ]);
```
Comment (nxthai23) [inline src/components/panels/shop/Shop.tsx:42]: check 2 store if needed => remove

File: src/components/panels/shop/Shop.tsx [hunk 2]
```diff
@@ -42,17 +57,51 @@ const Shop = () => {
...
+
+      if (response) {
+        setStaffs(response.cats);
+        setUser(response);
+        fetchStaffs();
+      }
+    } catch (error) {
+      console.error("Failed to buy item", error);
+    }
+  };
```
Comment (nxthai23) [inline src/components/panels/shop/Shop.tsx:93]: rename catConfig: catDeal => item:Item?

File: src/components/panels/shop/Shop.tsx [hunk 3]
```diff
@@ -65,12 +114,19 @@ const Shop = () => {
...
-  const { fetchCatDeals } = useFetchCatDeals();
+  const getItems = async () => {
+    try {
+      const response = await getCatDeals();
+      setCatDeals(response);
+      return response;
+    } catch (error) {
+      console.error("Failed to fetch cat deals", error);
+    }
+  };
```
Comment (nxthai23) [inline src/components/panels/shop/Shop.tsx:125]: rename => fetchItems()

File: src/components/ui/CardInfo.tsx [hunk 4]
```diff
@@ -93,7 +93,11 @@ const CardInfo: React.FC<Props> = ({ onBack, handleUpgrade }: Props) => {
                       <div className="flex justify-center mt-14 relative">
                         <div className="absolute bg-[#898989] w-[40%] h-2 rounded-[100%] left-1/2 -translate-x-1/2 bottom-3 z-30"></div>
                         <Image
-                          src={staff?.imgUrl || "/images/Cat.png"}
+                          src={
+                            staff?.imgUrl ||
+                            staff?.configId?.imgUrl ||
```
Comment (nxthai23) [inline src/components/ui/CardInfo.tsx:98]: why staff.configId? => check all

File: src/components/ui/CatCard.tsx [hunk 5]
```diff
@@ -5,19 +5,31 @@ import { get } from "lodash";
 import { Staff } from "@/types/common-types";
 
 type Props = {
-  cat: Staff | any;
+  cat?: Staff | any;
+  catDeal?: Staff | any;
```
Comment (nxthai23) [inline src/components/ui/CatCard.tsx:9]: remove catDeal here, this card only need cat object

File: src/components/ui/CatCard.tsx [hunk 6]
```diff
@@ -5,19 +5,31 @@ import { get } from "lodash";
...
+    imageUrl = get(catDeal.configId, "imgUrl", "");
+    name = get(catDeal.configId, "name", "");
+    numberStar = get(catDeal.configId, "numberStar", 0);
+    level = get(catDeal, "level", 0);
+  } else {
+    imageUrl = get(cat, "imgUrl", "");
+    name = get(cat, "name", "");
+    numberStar = get(cat, "numberStar", 0);
+    level = get(cat, "level", 0);
+  }
```
Comment (nxthai23) [inline src/components/ui/CatCard.tsx:32]: optimize

File: src/requests/shop/catDeal.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { BASE_URL } from "@/constants/api-url";
+import katAxios from "../axios.config";
+import { BuyBody, CatDeal } from "@/types/catDeal";
+
+export const getCatDeals = async () => {
+  const response = await katAxios.get<CatDeal[]>(`${BASE_URL}/shops`);
```
Comment (nxthai23) [inline src/requests/shop/catDeal.ts:6]: change CatDeal => Item

File: src/components/panels/manage/Manage.tsx [hunk 8]
```diff
@@ -217,6 +217,7 @@ const Manage: React.FC = () => {
   };
 
   const fetchDataUpgrade = async () => {
+    if (currentRestaurant?.level === 9) return;
```
[the remark this answers is not in the document]
Comment (hoanghongminh) [reply src/components/panels/manage/Manage.tsx:220]: fixed

File: src/components/ui/CatCard.tsx (hunk 5 above)
Comment (hoanghongminh) [reply src/components/ui/CatCard.tsx:9]: fixed

File: src/components/ui/CatCard.tsx (hunk 6 above)
Comment (hoanghongminh) [reply src/components/ui/CatCard.tsx:32]: fixed

File: src/components/panels/shop/Shop.tsx (hunk 1 above)
Comment (hoanghongminh) [reply src/components/panels/shop/Shop.tsx:42]: need because used in many components

File: src/components/panels/shop/Shop.tsx (hunk 2 above)
Comment (hoanghongminh) [reply src/components/panels/shop/Shop.tsx:93]: fixed

File: src/components/panels/shop/Shop.tsx (hunk 3 above)
Comment (hoanghongminh) [reply src/components/panels/shop/Shop.tsx:125]: fixed

File: src/components/ui/CardInfo.tsx (hunk 4 above)
Comment (hoanghongminh) [reply src/components/ui/CardInfo.tsx:98]: I fixed

File: src/requests/shop/catDeal.ts (hunk 7 above)
Comment (hoanghongminh) [reply src/requests/shop/catDeal.ts:6]: fixed

**Review by nxthai23 — APPROVED** [review]

**Pull request merged**